src/AbstractDiskWriter.cc

```diff
@@ -373,6 +373,14 @@ void AbstractDiskWriter::ensureMmapWrite(size_t len, int64_t offset)
         return;
       }
 
+      uint32_t filesize_lo = filesize & 0xffffffffu;
+      if (sizeof(off_t) > sizeof(size_t) && (filesize_lo) > 0) {
```

Comment:
You are right, I want to check if filesize is greater than max of size_t.

Your change looks more standardized. I will make the change.